includes/abstracts/abstract-batch.php

```diff
@@ -136,8 +143,47 @@
 	 */
 	public function get_results() {
 		$this->args = wp_parse_args( $this->args, $this->default_args );
+		$results = $this->batch_get_results();
 		$this->calculate_offset();
-		return $this->batch_get_results();
+		return $results;
+	}
+
+	/**
+	 * Set the total number of results
+	 *
+	 * Uses a number passed from the client to the server and compares it to the total objects
+	 * pulled by the latest query. If the dataset is larger, we increase the total_num_results number.
+	 * Otherwise, keep it at the original (to acount for deletion / changes).
+	 *
+	 * @param int $total_from_query Total number of results from latest query.
+	 */
+	public function set_total_num_results( $total_from_query ) {
+		// If this is past step 1, the client is passing back the total number of results.
+		// This accounts for deletion / descructive actions to the data.
+		$total_from_request = isset( $_POST['total_num_results'] ) ? absint( $_POST['total_num_results'] ) : 0; // Input var okay.
```

Comment:
Partially yes, this is more me identifying a potential refactoring to decouple this method from the $_POST global a bit. While reviewing the test I noticed how you have to set the $_POST value and it got me thinking of what a better way would be. This is not merge blocking.